### PR TITLE
MRG: Minor fixes to docstring and more copy params

### LINKED ISSUE #3070: Return self for in-place operations; Enable chaining for in-place operations
We have some inconsistencies about when we return `self` for in-place operations and when we don't. I think we apply this rule consistently when there is a `copy` param but it seems like we should have it elsewhere. it's useful for chaining. this was recently brought up on the listserv when someone tried to assign the `raw.filter()` output to a variable. If you wanted to `raw.filter().plot()` it wouldn't work.

### PR DESCRIPTION
**Pull request by Eric89GXL**
Lots of minor docstring fixes.

Ready for review/merge from my end.

Closes #3070.

### CODE REVIEW

File: mne/io/base.py [hunk 1]
```diff
@@ -921,44 +923,40 @@ def notch_filter(self, freqs, picks=None, filter_length='10s',
...
-                                  trans_bandwidth=trans_bandwidth,
-                                  method=method, iir_params=iir_params,
-                                  mt_bandwidth=mt_bandwidth, p_value=p_value,
-                                  picks=picks, n_jobs=n_jobs, copy=False)
+        _check_preload(raw, 'raw.notch_filter')
+        raw._data = notch_filter(
+            raw._data, fs, freqs, filter_length=filter_length,
+            notch_widths=notch_widths, trans_bandwidth=trans_bandwidth,
+            method=method, iir_params=iir_params, mt_bandwidth=mt_bandwidth,
+            p_value=p_value, picks=picks, n_jobs=n_jobs, copy=False)
```
Comment (jona-sassenhagen) [inline mne/io/base.py]: You're not actually returning raw are you?

Comment (Eric89GXL) [inline mne/io/base.py]: Hah!

**Comment by teonbrooks** [comment]
closes https://github.com/mne-tools/mne-python/issues/3070.

File: mne/io/base.py [hunk 2]
```diff
@@ -1837,6 +1862,14 @@ def _get_buffer_size(self, buffer_size_sec=None):
         return int(np.ceil(buffer_size_sec * self.info['sfreq']))
 
 
+def _check_preload(raw, msg):
+    """Helper to ensure data are preloaded"""
+    if not raw.preload:
+        raise RuntimeError(msg + ' requires raw data needs to be loaded. Use '
```
Comment (agramfort) [inline mne/io/base.py]: remove "needs"

**Comment by agramfort** [comment]
besides LGTM

merge when CIs are green

thx @Eric89GXL

**Comment by Eric89GXL** [comment]
Comments addressed, thanks. @jona-sassenhagen are you happy, too? If so I'll merge when CIs come back happy.

**Comment by jona-sassenhagen** [comment]
I don't have my laptop with me, so I can't do a good check, sorry. I can only try tomorrow.

**Comment by Eric89GXL** [comment]
@jona-sassenhagen that's fine

**Comment by jona-sassenhagen** [comment]
Looks good to me.

File: mne/filter.py [hunk 3]
```diff
@@ -1534,6 +1534,14 @@ def savgol_filter(self, h_freq):
             done using polynomial fits instead of FIR/IIR filtering.
             This parameter is thus used to determine the length of the
             window over which a 5th-order polynomial smoothing is used.
+        copy : bool
```
Comment (teonbrooks) [inline mne/filter.py]: I don't know if it matters or not but the copy description is a bit different given the function. should we have search/replace? we can also leave it as is

File: mne/io/base.py [hunk 4]
```diff
@@ -921,44 +928,53 @@ def notch_filter(self, freqs, picks=None, filter_length='10s',
...
+
+            .. versionadded:: 0.12.0
+
         verbose : bool, str, int, or None
             If not None, override default verbose level (see mne.verbose).
             Defaults to self.verbose.
 
+        Returns
+        -------
+        raw_filt : instance of Raw
```
Comment (teonbrooks) [inline mne/io/base.py]: `raw_filt` --> `raw`?

**Comment by teonbrooks** [comment]
other than those, LGTM, thanks @Eric89GXL

**Comment by teonbrooks** [comment]
other than that, LGTM. thanks @Eric89GXL

**Comment by agramfort** [comment]
LGTM

merge when you're happy

thx @Eric89GXL

File: mne/filter.py (hunk 3 above)
Comment (Eric89GXL) [inline mne/filter.py]: We can unify later if someone gets bored

**Comment by Eric89GXL** [comment]
Thanks for the reviews

**Pull request merged**